tests/frame/test_common.py

```diff
@@ -312,6 +313,30 @@ def test_join(df_raw: Any) -> None:
     compare_dicts(result_native, expected)
 
 
+@pytest.mark.parametrize(
+    "df_raw", [df_polars, df_lazy, df_pandas, df_pandas_na, df_lazy_na, df_mpd]
+)
+def test_cross_join(df_raw: Any) -> None:
+    df = nw.from_native(df_raw).select("a", "z")
+    result = df.join(df, how="cross", left_on=["a"], right_on="a")  # type: ignore[arg-type]
+
+    df_ = df.collect() if isinstance(df, nw.LazyFrame) else df
+
+    # List of dicts: [{"a": 1, "z": 7., ...}, {"a": 1, "z": 8., ...}, ...]
+    rows = [
+        {**left_row, **{k + "_right": v for k, v in right_row.items()}}
+        for left_row in df_.iter_rows(named=True)
+        for right_row in df_.iter_rows(named=True)
+    ]
+    # Pivot to single dict of values: {"a": [1, 1, ...], "z": [7., 8., ...], ...}
+    expected = defaultdict(list)
+    for item in rows:
+        for key, value in item.items():
+            expected[key].append(value)
```

Comment:
In my first attempt I just hardcoded some values, then I wanted to test for frames with NA values.

There could be a better way, but it did not come to my mind.

First step (`rows = .. `) generates the row values as list of dicts.
Second step, pivots those values in a single dict as expected from `compare_dicts`